homeassistant/components/kraken/sensor.py

```diff
@@ -187,6 +189,7 @@ def _try_get_state(self) -> str:
             return self.coordinator.data[self.tracked_asset_pair_wsname][
                 "opening_price"
             ]
+        raise TypeError
```

Comment:
This is definitely not `TypeError`. Maybe `KeyError`.
```suggestion
        raise HomeAssistantError(f"Unknown sensor type: {self._sensor_type}")
```